### DISCUSSION TITLE
Best Way To Append Status Since Custom Attribute Update

### DISCUSSION THREAD

**Discussion by nickpoulos** [spatie/laravel-model-status#110] [Q&A]
Since the update with the custom status attribute, I am no longer able to easily add `status` as an Eloquent Model append.   

I am currently using Laravel Model Status v1.14.0 and Laravel v9.52x.

I added the new `status_attribute` key in the config file array, which is set to the default value of `status`.  

My model has `status` added in the `$appends` class var.   

Previously, this was all that was needed, and the `status` attribute appears in my calls to `$model->toArray()` and `$model->toJson()`. In my particular case those calls are done behind the scenes with InertiaJs. 

However, since the update, the underlying method was removed, so I am now getting the error `Call to undefined method App\Models\Event::getStatusAttribute()`.  

Is there a new best way to handle status appends going forward? Or is this unintended behavior?

**Comment by chrisGeonet** [comment]
I've just ran into this issue.
I changed the status_attribute config value to "status_attribute", then added the following into each model that hasStatuses.
`	public function getStatusAttribute()
	{
		if($this->status()) {
			return $this->status_attribute;
		}else{
			return null;
		}
	}`

**Reply by nickpoulos** [reply]
This is still a thing as far as I am aware, and the above solution seems cumbersome/redundant since we are already including the HasStatuses trait in every model that requires it.  Yes we could wrap the HasStatuses trait in our own trait and just include that. 

But it seems like "appends" support would be something common this library should support out of the box. 

I am happy to attempt the PR if the Spatie team and other developers feel this is worth the effort to include.
